flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/delegation/PlannerBase.scala

```diff
@@ -384,6 +385,13 @@ abstract class PlannerBase(
         }
         val catalog = toScala(catalogManager.getCatalog(objectIdentifier.getCatalogName))
         val isTemporary = lookupResult.isTemporary
+
+        if (isManagedTable(catalog.get, resolvedTable) &&
+          !executor.isCheckpointingEnabled) {
+          throw new TableException(
+            s"Managed sink ${objectIdentifier} should enable the checkpointing.")
```

Comment:
`You should enable the checkpointing for managed table ${objectIdentifier}, managed table relies on checkpoint to commit and the data is visible only after commit.`
What do you think?